pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -144,7 +160,7 @@ func (s *ServiceEntryStore) workloadEntryHandler(old, curr model.Config, event m
 		s.deleteExistingInstances(key, instances)
 	}
 
-	s.edsUpdate(instances)
+	s.edsUpdate(append(instances, instancesDelete...))
```

Comment:
Why do we need deleted instance here?

Reply:
When the label changed, the endpoints of the cluster in Envoy is changed, so we need to send eds to Envoy.
`instancesDelete` is the list of clusters whose endpoints are changed.

Reply:
make sense. Almost forgot how we deal with it.